app/controllers/vm_cloud_controller.rb

```diff
@@ -113,16 +113,12 @@ def live_migrate
   def live_migrate_form_fields
     assert_privileges("instance_live_migrate")
     @record = find_by_id_filtered(VmOrTemplate, params[:id])
-    clusters = []
-    hosts = []
-    @record.ext_management_system.find_filtered_children("ems_clusters").each do |c|
-      clusters << {:id => c.id, :name => c.name}
+    clusters = @record.ext_management_system.ems_clusters.map do |c|
+      {:id => c.id, :name => c.name}
     end
-    @record.ext_management_system.find_filtered_children("hosts").each do |h|
-      hosts << {:id => h.id, :name => h.name, :cluster_id => h.emd_cluster.id}
+    hosts = @record.ext_management_system.hosts.map do |h|
```

Comment:
> we should always show just the filtered data. Otherwise as a user, you would see in your top level list of hosts, different data then in here

@mansam do you want to have another look/research at this possible issue before a BZ gets filed? I'm not sure how to filter :cry: